src/main/java/org/folio/service/FundsDistributionService.java

```diff
@@ -29,54 +30,64 @@ public class FundsDistributionService {
-          .orElseGet(() -> Money.zero(poLineCurrency));
+        .map(EncumbranceRelationsHolder::getFundDistribution)
+        .map(fundDistribution -> getDistributionAmount(fundDistribution, expectedTotal, poLineCurrency, conversion))
+        .reduce((money, money2) -> Money.from(MonetaryFunctions.sum(money, money2)))
+        .orElseGet(() -> Money.zero(poLineCurrency));
 
       MonetaryAmount remainder = expectedTotal.abs()
-          .subtract(calculatedTotal.abs());
+        .subtract(calculatedTotal.abs());
       int remainderSignum = remainder.signum();
       MonetaryAmount smallestUnit = getSmallestUnit(expectedTotal, remainderSignum);
 
-      for (ListIterator<EncumbranceRelationsHolder> iterator = getIterator(encumbranceRelationsHolders, remainderSignum); isIteratorHasNext(iterator, remainderSignum);) {
+      for (var iterator = getIterator(encumbranceRelationsHolders, remainderSignum); isIteratorHasNext(iterator, remainderSignum); ) {
 
-        final EncumbranceRelationsHolder holder = iteratorNext(iterator, remainderSignum);
-          CurrencyUnit fyCurrency = Monetary.getCurrency(holder.getCurrency());
-          MonetaryAmount initialAmount = getDistributionAmount(holder.getFundDistribution(), expectedTotal, poLineCurrency, conversion);
+        EncumbranceRelationsHolder holder = iteratorNext(iterator, remainderSignum);
+        CurrencyUnit fyCurrency = Monetary.getCurrency(holder.getCurrency());
+        MonetaryAmount initialAmount = getDistributionAmount(holder.getFundDistribution(), expectedTotal, poLineCurrency, conversion);
 
-          if (FundDistribution.DistributionType.PERCENTAGE.equals(holder.getFundDistribution().getDistributionType()) && !remainder.isZero()) {
-            initialAmount = initialAmount.add(smallestUnit);
-            remainder = remainder.abs().subtract(smallestUnit.abs()).multiply(remainderSignum);
-          }
+        if (holder.getFundDistribution().getDistributionType() == FundDistribution.DistributionType.PERCENTAGE && !remainder.isZero()) {
+          initialAmount = initialAmount.add(smallestUnit);
+          remainder = remainder.abs().subtract(smallestUnit.abs()).multiply(remainderSignum);
+        }
 
-          MonetaryAmount expended = Optional.of(holder).map(EncumbranceRelationsHolder::getNewEncumbrance).map(Transaction::getEncumbrance).map(Encumbrance::getAmountExpended).map(aDouble -> Money.of(aDouble, fyCurrency))
-              .orElse(Money.zero(fyCurrency));
-          MonetaryAmount awaitingPayment = Optional.of(holder).map(EncumbranceRelationsHolder::getNewEncumbrance).map(Transaction::getEncumbrance).map(Encumbrance::getAmountAwaitingPayment)
-              .map(aDouble -> Money.of(aDouble, fyCurrency)).orElse(Money.zero(fyCurrency));
-          MonetaryAmount amount = calculateEncumbranceEffectiveAmount(initialAmount, expended, awaitingPayment, fyCurrency);
+        MonetaryAmount expended = getOrZero(holder, fyCurrency, FinanceUtils::getExpended);
```

Comment:
I believe there shuold be credits amount as well. Damien once said that wherever you see expended, you should include credits as well

Reply:
`FinanceUtils::getExpended` is exactly what we need - it takes `expended - credited`. Created utility method to not copy/paste the logic

Reply:
getExpended should return expended. getCredited should return credited. https://github.com/folio-org/mod-finance-storage/pull/414#discussion_r1638310685

Reply:
Maybe name is confusing. If you double check this we use this expended amount to calculate encumbrance effective amount: `calculateEncumbranceEffectiveAmount`: 
Previously it was: `Math.max(initialAmount - expended - awaitingPayment, 0)`
Now it is `Math.max(initialAmount - expended + *credited* - awaitingPayment, 0)`
I will update the code so that get `ExpendedAmount` will be taken and instead update calculations in `calculateEncumbranceEffectiveAmount`